hnn_core/cell.py

```diff
@@ -250,13 +250,22 @@ def _create_sections(self, p_secs, topology):
         for height and xz plane for depth. This is opposite for model as a
         whole, but convention is followed in this function ease use of gui.
         """
+        # distance from initial to final root postion
+        # Resolve: y-coordinate here corresponds to the z-coordinate in
+        # self.pos[2]
+        dx = self.pos[0] - self.p_secs['soma']['sec_pts'][0][0]
+        dy = self.pos[2] - self.p_secs['soma']['sec_pts'][0][1]
+        dz = self.pos[1] - self.p_secs['soma']['sec_pts'][0][2]
+
         for sec_name in p_secs:
             sec = h.Section(name=f'{self.name}_{sec_name}')
             self.sections[sec_name] = sec
 
             h.pt3dclear(sec=sec)
             for pt in p_secs[sec_name]['sec_pts']:
-                h.pt3dadd(pt[0], pt[1], pt[2], 1, sec=sec)
+                h.pt3dadd(pt[0] + dx,
+                          pt[1] + dy,
+                          pt[2] + dz, 1, sec=sec)
```

Comment:
The consequences of these lines are very confusing to me (still: I realise nothing has changed in that respect in this PR). Are we reorienting the apical sections to align with the Z-axis here?

In any case, we now have a situation where `net.pos_dict`, `Cell.pos` and the `Section` end points are inconsistent, right? 

@rythorpe are you planning to address this in the follow-up you mention? I'm working on some qualified opinions about the spacing-issue too, based on "reduced" model literature.

Reply:
yes, we are reorienting the apical sections to align with the Z-axis ... see the lines above.

The `Section` end points are specified relative to the `Cell.pos` or `net.pos_dict` ...

Reply:
now that I think of it, to reorient along the Z-axis, you'd need to do:

```py
dy = self.pos[1] - self.p_secs['soma']['sec_pts'][0][2]
```

what might be more clear though is to apply a `trans` matrix first:

```py
trans = [[1, 0, 0], [0, 0, 1], [0, 1, 0]]
self.p_secs['soma']['sec_pts'] = np.dot(trans, self.p_secs['soma']['sec_pts'])
```

there is a wealth of functions we can copy from MNE :) There is even a function to align along a target axis:

https://github.com/mne-tools/mne-python/blob/b3f4b05b3933a3b73458205c7e786166f90ad1f7/mne/transforms.py#L298 

that we can use to our advantage

Reply:
I'm not sure I understand your question @cjayb. Are you asking about the implementation in this PR or what we plan to do in later PRs?

If we change the section endpoints in `params_default.py`, I think we should do that in a separate PR.